src/fscacher/cache.py

```diff
@@ -76,13 +77,24 @@ def memoize_path(self, f):
         # we need to actually decorate a function
         fingerprint_kwarg = "_cache_fingerprint"
 
-        @self.memoize
         @wraps(f)  # important, so .memoize correctly caches different `f`
         def fingerprinted(path, *args, **kwargs):
             _ = kwargs.pop(fingerprint_kwarg)  # discard
             lgr.debug("Running original %s on %r", f, path)
             return f(path, *args, **kwargs)
 
+        # We need to add the fingerprint_kwarg to fingerprinted's signature so
+        # that joblib doesn't complain:
+        sig = signature(fingerprinted)
+        fp_kwarg_param = Parameter(
+            fingerprint_kwarg, Parameter.KEYWORD_ONLY, default=None
+        )
+        sig2 = sig.replace(
+            parameters=tuple(sig.parameters.values()) + (fp_kwarg_param,)
```

Comment:
just to make sure -- `.parameters.values()` should be consistently (across runs) ordered in the versions of python we support, right?

Reply:
Yes, it's documented as ["an ordered mapping"](https://docs.python.org/3.6/library/inspect.html#inspect.Signature.parameters) all the way back to at least 3.6.